app/models/rbac.rb

```diff
@@ -161,17 +171,13 @@ def self.combine_filtered_ids(u_filtered_ids, b_filtered_ids, m_filtered_ids, d_
   end
 
   def self.find_targets_with_indirect_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
-    parent_class   = rbac_class(klass)
-    u_filtered_ids = get_self_service_object_ids(user_or_group, parent_class)
-    b_filtered_ids = get_belongsto_filter_object_ids(parent_class, rbac_filters['belongsto'])
-    m_filtered_ids = get_managed_filter_object_ids(parent_class, parent_class, rbac_filters['managed'])
-    d_filtered_ids = rbac_filters['ids_via_descendants']
-    filtered_ids   = combine_filtered_ids(u_filtered_ids, b_filtered_ids, m_filtered_ids, d_filtered_ids)
+    parent_class = rbac_class(klass)
+    filtered_ids = calc_filtered_ids(parent_class, parent_class, rbac_filters, user_or_group).first
 
     find_targets_filtered_by_parent_ids(parent_class, klass, scope, find_options, filtered_ids)
   end
 
-  def self.compute_total_count(klass, scope, extra_target_ids, conditions, includes = nil)
+  def self.total_scope(klass, scope, extra_target_ids, conditions, includes)
```

Comment:
`klass` is unused (a la rubocop)...can you just drop the parameter? (can be separate PR so as not to scope creep)